proto/pocket/service/tx.proto

```diff
@@ -10,7 +15,9 @@ service Msg {
 }
 
 message MsgAddService {
-  string address = 1;
+  option (cosmos.msg.v1.signer) = "supplier_address"; // https://docs.cosmos.network/main/build/building-modules/messages-and-queries
```

Comment:
I've had some time to think about this and think that this should not be the `supplier`.

I'm documenting it in this thread so we can capture the conversation and easily transfer it to the documentation in the future, but lmk if we should jump on a call to reduce the feedback loop.

**We have two goals:**
1. Permissionless Services
2. Sybil Attack Prevention

**1. Permissionless Services**
	- Any address (supplier or not) should be able add a service
	- If a new service is added that doesn't have suppliers, I think we should simply update "session 	hydration" to return an error

**2. Sybil Attack Prevention**
	- Allow anyone to create services, but prevent sybil attacks or "on-chain DoS"
	- Charge individuals for adding a new services (more money to "the chain"" and prevent disincentive)

This creates a "separation of concerns" by decoupling who creates services and who supports them. There's an opportunity to create interesting marketing dynamics.

**Suggested changes before further review:**
1. Change `supplier_address` to just `address` which is the signer
2. Add a gov parameter call `add_service_fee` and fail if `address` can't pay it
3. Transfer funds from `address` to the `service_module_acc`

Reply:
Yeah to be clear this can actually be any actor - its not enforced to be a `supplier` staked actor at all. The naming confusion is a concern for sure - will get into the changes ASAP. I am adding the governance param in a followup PR so that won't come in this so points 2,3 will come in a seperate PR which includes both a governance param and fee deduction. Or else this would be a very large PR. Will address point 1 in this PR and keep 2 and 3 for the next one thats in flight.

Reply:
> Yeah to be clear this can actually be any actor - its not enforced to be a supplier staked actor at all. The naming confusion is a concern for sure - will get into the changes ASAP. I 

👍 

> I am adding the governance param in a followup PR so that won't come in this so points 2,3 will come in a seperate PR which includes both a governance param and fee deduction

Sounds good. 

This is a pretty key/important feature. Do you mind creating a ticket for it?

I think that even taking a screenshot of this thread + a `TODO_BLOCKER(#XXX)`

Reply:
The PR will be up tmrw I don't see the point adding a todo when the PR will be up soon. I just need to update tests for the message handler and mock of the service keeper to be able to handle the new logic its in the branch `feat/add-service-fee`

Reply:
Saw that you did end up adding the TODO and appreciate it!